src/pages/POS/POS.vue

```diff
@@ -971,6 +997,122 @@ export default defineComponent({
         });
       }
     },
+    async handleBatchSelected(batchName: string) {
+      if (!this.pendingBatchItem) {
+        return;
+      }
+
+      const { item, quantity } = this.pendingBatchItem;
+      this.pendingBatchItem = null;
+
+      try {
+        const isInventoryItem = await this.fyo.getValue(
+          ModelNameEnum.Item,
+          item.name,
+          'trackItem'
+        );
+        if (!this.batchAddedItems.includes(item.name)) {
+          this.batchAddedItems.push(item.name);
+        }
+
+        let availableQty = 0;
+        if (isInventoryItem) {
+          availableQty =
+            (await fyo.db.getStockQuantity(
+              item.name,
```

Comment:
```suggestion
        const itemDoc = (await this.fyo.doc.getDoc(
          ModelNameEnum.Item,
          item.name
        )) as Item;

        let availableQty = 0;
        if (itemDoc.trackItem) {
          availableQty =
            (await fyo.db.getStockQuantity(
              item.name,
```
<br/>
You are already calling getDoc for this item below, so you can use itemDoc here instead of getValue